frontend/src/container/FormAlertRules/index.tsx

```diff
@@ -102,6 +103,13 @@ function FormAlertRules({
 	const [alertDef, setAlertDef] = useState<AlertDef>(initialValue);
 	const [yAxisUnit, setYAxisUnit] = useState<string>(currentQuery.unit || '');
 
+	useEffect(() => {
+		if (!isEqual(currentQuery.unit, yAxisUnit)) {
```

Comment:
didn't add conditional check here for currentQuery.unit as the default for that is undefined hence we need `''` in that case